### PR TITLE
Create marketer relationship with access and discount codes

### LINKED ISSUE #4093: Create marketer relationship with access_code and discount_code api
Endpoints:
- `/access-codes/<access_code_id/marketer` (detail)
-  `/discount-codes/<access_code_id/marketer` (detail)
- `/access-codes/<access_code_id/relationships/marketer` (relationship)
- `/discount-codes/<access_code_id/relationships/marketer` (relationship)
- ~~`/marketer/<user_id/access-codes` (list)~~
- ~~`/marketer/<user_id/discount-codes` (list)~~
- ~~`/marketer/<user_id/relationships/discount-codes` (relationship)~~
- ~~`/marketer/<user_id/relationships/access-codes` (relationship)~~

### PR DESCRIPTION
**Pull request by magdalenesuo**
Fixes #4093 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `nextgen` branch.
- [X] The unit tests pass locally with my changes 
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Create marketer relationship with access and discount codes

#### Changes proposed in this pull request:

-
-
-

### CODE REVIEW

File: app/api/__init__.py [hunk 1]
```diff
@@ -43,11 +43,12 @@
 api.route(UserDetail, 'user_detail', '/users/<int:id>', '/notifications/<int:notification_id>/user',
           '/event-invoices/<int:event_invoice_id>/user', '/users-events-roles/<int:users_events_role_id>/user',
           '/speakers/<int:speaker_id>/user', '/access-codes/<int:access_code_id>/user',
-          '/email-notifications/<int:email_notification_id>/user')
+          '/email-notifications/<int:email_notification_id>/user', '/discount-codes/<int:discount_code_id>/user')
```
Comment (niranjan94) [inline app/api/__init__.py]: Should be called marketer here. `/discount-codes/<int:discount_code_id>/<relationship-name>`

Comment (magdalenesuo) [inline app/api/__init__.py]: @niranjan. done

**Comment by magdalenesuo** [comment]
@niranjan94 Made the changes. Please review

File: app/api/users.py [hunk 2]
```diff
@@ -83,13 +82,21 @@ class Meta:
         many=True,
         type_='speaker')
     access_codes = Relationship(
-        attribute='access_codes',
+        attribute='access_code',
```
Comment (niranjan94) [inline app/api/users.py]: If the relationship is plural, ensure attribute name is also plural

File: app/api/users.py [hunk 3]
```diff
@@ -83,13 +82,21 @@ class Meta:
...
         self_view='v1.user_access_codes',
         self_view_kwargs={'id': '<id>'},
         related_view='v1.access_code_list',
         related_view_kwargs={'user_id': '<id>'},
         schema='AccessCodeSchema',
         type_='access-codes')
+    discount_codes = Relationship(
+        attribute='discount_code',
```
Comment (niranjan94) [inline app/api/users.py]: If the relationship is plural, ensure attribute name is also plural

Comment (magdalenesuo) [inline app/api/users.py]: Done.

File: app/api/users.py (hunk 2 above)
Comment (magdalenesuo) [inline app/api/users.py]: Done

**Pull request merged**